ops/testing.py

```diff
@@ -2503,6 +2521,14 @@ def _check_protocol_and_port(self, protocol: str, port: Optional[int]):
         else:
             raise model.ModelError(f'ERROR invalid protocol "{protocol}", expected "tcp", "udp", or "icmp"\n')  # NOQA: test_quote_backslashes
 
+    def reboot(self, now: bool = False):
+        if not now:
+            # We can't simulate the reboot, so just do nothing.
```

Comment:
While we're considering testing of this, I think we should somehow signal this case so that it can be tested for -- that reboot was called (without now).

My suggesting would be we set a "rebooted" flag (whether it was called with now or not) and then add a Harness method like this:

```python
def rebooted(self) -> bool:
    """Report whether the charm code called :meth:`Unit.reboot` (with or without *now*)."""
```

Reply:
I actually had something similar to this in an earlier version, and ripped it out when I gave up trying to mangle harness into magically re-emitting the event with `now=True`. Bringing it back sounds good to me.

My preference would be for it to return a datetime.datetime indicating when the last (call to) reboot was and `None` if there hasn't been one. It feels like this makes it easier to test for behaviour when there are multiple reboots expected - if it's a bool then I think we would really need to provide a way to reset the flag. An int counter would also work if you'd rather not use a datetime. With either you can still do a plain `self.assertTrue()` if all you care about is that it was rebooted at least once.

I've added the datetime version - lmk if you'd prefer changes.

Reply:
Good idea, thanks!